packages/model-viewer/src/features/ar.ts

```diff
@@ -327,8 +334,15 @@ configuration or device capabilities');
      * Takes a URL to a USDZ file and sets the appropriate fields so that Safari
      * iOS can intent to their AR Quick Look.
      */
-    [$openIOSARQuickLook]() {
-      const modelUrl = new URL(this.iosSrc!, self.location.toString());
+    async[$openIOSARQuickLook]() {
+      if (this.arAutogenerateUsdz && !this[$generatedIosUrl]) {
```

Comment:
Ah, I just noticed a problematic aspect of your feature 2: if the file changes due to the scene-graph API (switched variant, swapped texture/color etc) and you jump into Quick Look again, you won't see the changes due to caching the USDZ. We could either skip the cache entirely (I don't think the perf change is a huge deal, especially since going into Quick Look twice with the same model is probably pretty rare), or else we'd have to add some kind of `isDirty` for the scene-graph functions that would tell us to regenerate the USDZ. I think simpler is better, especially for the first version.

Reply:
Yeah, but the chance of breaking things is high. Again, I don't think the perf gain here is enough to justify the complexity. Let's just skip the USDZ caching and call it a win on memory footprint. 😁

Reply:
alright, for now, let's make those bionic chips do their job :). I really hope they will finally let us use WebXR on iOS directly so we can throw away all this for good :)